tycho-indexer/src/services/rpc.rs

```diff
@@ -1233,6 +1426,134 @@ mod tests {
+            entry_point: entry_point.clone(),
+            params: tracing_params_a.clone(),
+        };
+        let entry_point_with_params_b = EntryPointWithTracingParams {
+            entry_point: entry_point.clone(),
+            params: tracing_params_b.clone(),
+        };
+
+        let trace_result_a = TracingResult {
+            retriggers: HashSet::from([(
+                Bytes::from("0x00000000000000000000000000000000000000aa"),
+                Bytes::from("0x0000000000000000000000000000000000000aaa"),
+            )]),
+            called_addresses: HashSet::from([Bytes::from(
+                "0x0000000000000000000000000000000000aaaa",
+            )]),
+        };
+        let trace_result_b = TracingResult {
+            retriggers: HashSet::from([(
+                Bytes::from("0x00000000000000000000000000000000000000bb"),
+                Bytes::from("0x0000000000000000000000000000000000000bbb"),
+            )]),
+            called_addresses: HashSet::from([Bytes::from(
+                "0x0000000000000000000000000000000000bbbb",
+            )]),
+        };
+        let params_to_trace_result = HashMap::from([
+            (tracing_params_a.clone(), trace_result_a.clone()),
+            (tracing_params_b.clone(), trace_result_b.clone()),
+        ]);
+
+        let expected_entry_points_with_params = HashMap::from([(
+            component_id.clone(),
+            HashSet::from([entry_point_with_params_a.clone(), entry_point_with_params_b.clone()]),
+        )]);
+
+        let expected_trace_results =
+            HashMap::from([(entry_point_id.clone(), params_to_trace_result)]);
+
+        let mut gw = MockGateway::new();
+
+        let mock_get_entry_points_response =
+            Ok(WithTotal { entity: expected_entry_points_with_params.clone(), total: Some(3) });
+        gw.expect_get_entry_points_tracing_params()
+            .return_once(|_, _| Box::pin(async move { mock_get_entry_points_response }));
+
+        let mock_traced_entry_points_response = Ok(expected_trace_results.clone());
+        gw.expect_get_traced_entry_points()
+            .return_once(|_| Box::pin(async move { mock_traced_entry_points_response }));
+
+        let req_handler = RpcHandler::new(gw, None);
+
+        // request for 2 entry points that are in the DB
+        let request = dto::TracedEntryPointRequestBody {
+            chain: dto::Chain::Ethereum,
+            protocol_system: "uniswap_v2".to_string(),
+            component_ids: Some(vec![component_id.clone()]),
+            pagination: dto::PaginationParams { page: 0, page_size: 2 },
+        };
+
+        // First request
+        let traced_entry_points = req_handler
+            .get_traced_entry_points(&request)
+            .await
+            .unwrap();
+
+        assert_eq!(
+            traced_entry_points
+                .traced_entry_points
+                .get(&component_id)
+                .unwrap()
+                .len(),
+            2
+        );
```

Comment:
Didn't test for actual contents here because I saw a few other tests don't do it, so I wasn't sure if we just test for this more thoroughly at a different level? 

I can add a more thorough assert tomorrow (ignoring the ordering inside the returned vec) though if you think it's important!